src/main/scala-2.12/org/ergoplatform/appkit/ergotool/dex/CancelOrderCmd.scala

```diff
@@ -108,69 +108,81 @@ object CancelOrder {
-               unspentBoxesForAmount: (Long) => Seq[InputBox]): Try[TxProto] =
-    outBoxProto(orderBox, recipientAddress)
-      .map { outbox =>
-        val inputBoxes = selectInputBoxes(orderBox, outbox.getValue, unspentBoxesForAmount)
-        TxProto(inputBoxes, Seq(outbox), MinFee, recipientAddress)
-      }
+               unspentBoxesForAmount: (Long) => Seq[InputBox]): TxProto = {
+    val outbox = outBoxProto(orderBox, recipientAddress)
+    val inputBoxes = selectInputBoxes(orderBox, outbox.getValue, unspentBoxesForAmount)
+    TxProto(inputBoxes, Seq(outbox), MinFee, recipientAddress)
+  }
 
-  def outBoxProto(orderBox: InputBox, recipientAddress: Address): Try[OutBoxProto] = {
-    Try {
-      val orderBoxContractTemplate = ErgoTreeTemplate.fromErgoTree(orderBox.getErgoTree)
-      val orderBoxId = orderBox.getId
-      val txFee = MinFee
-      val outboxContract = new ErgoTreeContract(SigmaPropConstant(recipientAddress.getPublicKey))
-      // TODO: add ErgoTreeTemplate.equals
-      if (util.Arrays.equals(orderBoxContractTemplate.getBytes, SellerContract.contractTemplate.getBytes)) {
-        // sell order
-        val sellerPk = SellerContract.sellerPkFromTree(orderBox.getErgoTree)
-          .getOrElse(sys.error(s"cannot extract seller PK from order box $orderBoxId"))
-        require(sellerPk == recipientAddress.getPublicKey,
-          s"sell order box $orderBoxId can be claimed with $sellerPk PK, while your's is ${recipientAddress.getPublicKey}")
-        OutBoxProto(orderBox.getValue - txFee, Seq(orderBox.getTokens.get(0)), outboxContract)
-      } else if (util.Arrays.equals(orderBoxContractTemplate.getBytes, BuyerContract.contractTemplate.getBytes)) {
-        // buy order
-        val buyerPk = BuyerContract.buyerPkFromTree(orderBox.getErgoTree)
-          .getOrElse(sys.error(s"cannot extract buyer PK from order box $orderBoxId"))
-        require(buyerPk == recipientAddress.getPublicKey,
-          s"buy order box $orderBoxId can be claimed with ${buyerPk} PK, while yours is ${recipientAddress.getPublicKey}")
-        // as a workaround for https://github.com/ScorexFoundation/sigmastate-interpreter/issues/628
-        // box.tokens cannot be empty
-        val token = new ErgoToken("21f84cf457802e66fb5930fb5d45fbe955933dc16a72089bf8980797f24e2fa1",
-          0L)
-        OutBoxProto(orderBox.getValue - txFee, Seq(token), outboxContract)
-      } else {
-        sys.error(s"unsupported contract type in box ${orderBoxId.toString}")
-      }
+  def outBoxProto(orderBox: InputBox, recipientAddress: Address): OutBoxProto = {
+    val orderBoxContractTemplate = ErgoTreeTemplate.fromErgoTree(orderBox.getErgoTree)
+    val orderBoxId = orderBox.getId
+    val txFee = MinFee
+    val outboxContract = new ErgoTreeContract(SigmaPropConstant(recipientAddress.getPublicKey))
+    // TODO: add ErgoTreeTemplate.equals
+    if (util.Arrays.equals(orderBoxContractTemplate.getBytes, SellerContract.contractTemplate.getBytes)) {
```

Comment:
@greenhat looks like unreasonable technical debt. Let's fix it.

Reply:
Done. Corresponding appkit's PR - https://github.com/aslesarenko/ergo-appkit/pull/36

Reply:
Merged https://github.com/aslesarenko/ergo-appkit/pull/36